### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Ilanak**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "type": "git",
+    "url": "git+https://github.com/CatalystCode/S3-to-adl-data-transfer.git"
+  },
+  "keywords": [
+    "DataLake",
+    "S3",
+    "Azure"
+  ],
+  "author": "ilanak",
+  "license": "ISC",
```
Comment (torosent) [inline package.json]: MIT

File: package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "s3-to-adl-data-transfer",
+  "version": "1.0.0",
```
Comment (torosent) [inline package.json]: better start with 0.0.1

File: package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "s3-to-adl-data-transfer",
```
Comment (torosent) [inline package.json]: let's change then name to a shorter one to comply with npmjs like "s3toadl"

File: package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  "dependencies": {
+    "async": "^2.5.0",
+    "aws-sdk": "^2.88.0",
+    "azure-arm-datalake-store": "^2.1.0-preview",
+    "fs": "^0.0.1-security",
+    "rimraf": "^2.6.1",
+    "underscore": "^1.8.3"
+  },
+  "devDependencies": {},
+  "scripts": {
```
Comment (torosent) [inline package.json]: remove

File: azureDataLakeModule.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    console.log("file: " + fileFullName + " exists in data lake");
+
+    // If file exist in Azure Data Lake but it's been updated in aws - upload it again
+    if (file.fileStatus.modificationTime < awsFile.LastModified.getTime()) {
+      return true;
+    }
+
+    return false;
+  }).catch((ex) => {
+    console.log("file: " + fileFullName + " doesn't exists in ADL");
```
Comment (torosent) [inline azureDataLakeModule.js]: Use logging framework like winston.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (torosent) [inline package.json]: add tests or remove

File: azureDataLakeModule.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+var accountName = config.azureDataLakeAccountName;
+
+// Checks if aws file exists in ADL, or if S3 holds a newer version of file
+exports.shouldUploadToADL = function (awsFile) {
+  var fileFullName = awsFile.Key;
+  return filesystemClient.fileSystem.getFileStatus(accountName, fileFullName).then(file => {
+    console.log("file: " + fileFullName + " exists in data lake");
+
+    // If file exist in Azure Data Lake but it's been updated in aws - upload it again
+    if (file.fileStatus.modificationTime < awsFile.LastModified.getTime()) {
```
Comment (torosent) [inline azureDataLakeModule.js]: We need to check the time changes. AWS time might be different than Azure

Comment (Ilanak) [inline azureDataLakeModule.js]: I checked it locally, it looks the same time..

**Pull request merged**